src/App.tsx

```diff
@@ -9,52 +10,192 @@
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [users, setUsers] = useState<User[]>([]);
+  const [isLoadingUsers, setIsLoadingUsers] = useState(false);
+  const [isLoadingPosts, setIsLoadingPosts] = useState(false);
+  const [isLoadingComments, setIsLoadingComments] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isFormOpen, setIsFormOpen] = useState(false);
+
+  useEffect(() => {
+    const fetchUsers = async () => {
+      setIsLoadingUsers(true);
+      try {
+        const usersData = await getUsers();
+
+        setUsers(usersData);
+        setError(null);
+      } catch (err) {
+        setError('Failed to load users');
+      } finally {
+        setIsLoadingUsers(false);
+      }
+    };
+
+    fetchUsers();
+  }, []);
+
+  useEffect(() => {
+    const fetchPosts = async () => {
+      setIsLoadingPosts(true);
+
+      try {
+        if (selectedUser) {
+          const userPosts = await getPosts(selectedUser.id);
+
+          setPosts(userPosts);
+          setError(null);
+        }
+      } catch (err) {
+        setError('Failed to load posts');
+      } finally {
+        setIsLoadingPosts(false);
+      }
+    };
+
+    fetchPosts();
+  }, [selectedUser]);
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  useEffect(() => {
+    const fetchComments = async () => {
+      setIsLoadingComments(true);
+      try {
+        if (selectedPost) {
+          const postComments = await getComments(selectedPost.id);
 
-              <Loader />
+          setComments(postComments);
+          setError(null);
+        }
+      } catch (err) {
+        setError('Failed to load comments');
+      } finally {
+        setIsLoadingComments(false);
+      }
+    };
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
+    fetchComments();
+  }, [selectedPost]);
+
+  const handleDeleteComment = async (commentId: number) => {
+    try {
+      await deleteComment(commentId);
+      setComments(currentComments =>
+        currentComments.filter(comment => comment.id !== commentId),
+      );
+    } catch (err) {
+      setError('Unable to delete comment');
+    }
+  };
+
+  const handleAddComment = async (comment: Comment) => {
+    try {
+      const newComment: Comment = await addComment({
+        id: comment.id,
+        postId: comment.postId,
+        name: comment.name,
+        email: comment.email,
+        body: comment.body,
+      });
+
+      setComments(prevComments => [...prevComments, newComment]);
+    } catch (err) {
+      setError('Unable to add a comment');
+      throw err;
```

Comment:
Throwing an error after setting the error state in `handleAddComment` might not be necessary unless you have a specific reason to propagate the error further. Consider removing `throw err;` if it's not needed, as it could lead to unhandled promise rejections.